### ISSUE TITLE
Question: is it possible define vpc resources and share it across all workspaces

### ISSUE THREAD

**Issue by KursLabIgor** [hashicorp/terraform#21392]
Terraform v0.11.11.
I have main.tf file where via modules defined a lot of resources. Remote backend has partial configuration
In my case i have development environment with own workspace where defined all resources including VPC. 
But VPC resources could be shared across all type workspace inside one region. 
I cant figure out how to achieve it when we do not specify bucket name in tf files and when i have partial backend config. I dont know what bucket name will be until Make file created it. 
This How I create s3 backend.
```
./aws-vault exec $(ENV) -- aws s3api create-bucket \
			--bucket $(S3_BUCKET) \
			--acl private \
			--region $(REGION) \
			--create-bucket-configuration LocationConstraint=$(REGION); \
./aws-vault exec $(ENV) -- aws s3api put-bucket-versioning \
			--bucket $(S3_BUCKET) \
			--versioning-configuration Status=Enabled; \
```
So basically idea following
s3_remote_state_bucket
                |_vpc.tfstate
                |_workspace1
                |_workspaceN

I hope my case will be clear for understanding.

**Comment by KursLabIgor** [comment]
question is not relevant anymore. I decided move all infra stuff to separate terraform project. And use remote_state to access to vpc stuff.
